app/actions/watchlist.ts

```diff
@@ -0,0 +1,81 @@
+import { AppDispatch, GetState } from "../store";
+import fetchWatchlist from "../utils/fetchWatchlist";
+
+export const WATCHLIST_COMPLETE = "WATCHLIST_COMPLETE";
+export const WATCHLIST_ERROR = "WATCHLIST_ERROR";
+export const WATCHLIST_LOADING = "WATCHLIST_LOADING";
+
+const loading = () => {
+  return {
+    type: WATCHLIST_LOADING,
+  };
+};
+
+const error = () => {
+  return {
+    type: WATCHLIST_ERROR,
+  };
+};
+
+// TODO: remove any type
+const complete = (data: any) => {
+  return {
+    type: WATCHLIST_COMPLETE,
+    data: data,
+  };
+};
+
+export const add = (movie: any) => async (
```

Comment:
`movies` is probably the only `any` left in the whole project. Is another item for the missing/todo list, and would need to be applied around all code.